pkg/progress/progress.go

```diff
@@ -144,32 +147,37 @@ func (b *terminalProgressBar) SetProgress(subLevel int, msg string, done int, to
 	return nil
 }
 
-func shorten(msg string) string {
-	msg = strings.Replace(msg, "\n", " ", -1)
-	// XXX: make this smarter
-	if len(msg) > 60 {
-		return msg[:60] + "..."
+func getTerminalWidth() int {
+	width, _, err := term.GetSize(int(os.Stdout.Fd()))
+	if err != nil {
```

Comment:
It might be safer to assume that the terminal is in fact broken if it's not responding correctly to `TIOCGWINSZ` and instead downgrade to a non-interactive terminal.

By that I mean that *if* progress is set to auto we currently only check if isatty returns true on the fd; we should probably call GetSize there as well to see if that terminal is behaving correctly.